main.tf

```diff
@@ -84,22 +50,119 @@ module "password_lambda" {
         "ssm:DeleteParameter"
       ]
       resources = [
-        #should be variable passed in per instance.
         "arn:${data.aws_partition.current.partition}:ec2:${var.region}:${data.aws_caller_identity.current.account_id}:instance/*",
         "arn:${data.aws_partition.current.partition}:ssm:${var.region}:${data.aws_caller_identity.current.account_id}:*",
         "arn:${data.aws_partition.current.partition}:ssm:${var.region}::document/AWS-RunShellScript",
         "arn:${data.aws_partition.current.partition}:ssm:${var.region}::document/AWS-RunPowerShellScript"
       ]
-    },
+    }
+  } : null
+  policy_statements_log_exportation = var.which_lambda_function == "log_exportation" ? {
+    "a" = {
+      effect = "Allow"
+      actions = [
+        "logs:CreateExportTask",
+        "logs:Describe*",
+        "logs:ListTagsLogGroup"
+      ]
+      resources = ["*"]
+    }
+    "b" = {
+      effect = "Allow"
+      actions = [
+        "ssm:DescribeParameters",
+        "ssm:GetParameter",
+        "ssm:GetParameters",
+        "ssm:GetParametersByPath",
+        "ssm:PutParameter"
+      ]
+      resources = ["arn:${data.aws_partition.current.partition}:ssm:${var.region}:${data.aws_caller_identity.current.account_id}:parameter/log-exporter-last-export/*"]
+    }
+    "c" = {
+      effect = "Allow"
+      actions = [
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ]
+      resources = ["arn:${data.aws_partition.current.partition}:logs:${var.region}:${data.aws_caller_identity.current.account_id}:*"]
+    }
+    "d" = {
+      effect = "Allow"
+      actions = [
+        "s3:PutObject",
+        "s3:PutObjectACL"
+      ]
+      resources = ["arn:${data.aws_partition.current.partition}:s3:::${var.cloudwatch_logs_export_bucket}"]
+    }
+    "e" = {
+      effect = "Allow"
+      actions = [
+        "s3:PutBucketAcl",
+        "s3:GetBucketAcl"
+      ]
+      resources = ["arn:${data.aws_partition.current.partition}:s3:::${var.cloudwatch_logs_export_bucket}"]
+    }
+  } : null
+
+  dynamic_environment_variables = var.which_lambda_function == "password_rotation" ? {
+    users        = join(",", var.users)
+    instance_ids = join(",", var.instance_ids)
+    } : var.which_lambda_function == "log_exportation" ? {
+    cloudwatch_logs_export_bucket = var.cloudwatch_logs_export_bucket
+  } : {}
```

Comment:
I feel like this might get annoying to scale as we add more functions using this ternary pattern.
I'd recommend moving this to multiple locals (one local for `password_rotation_environment_variables`, one local for `log_exportation_environment_variables`) and then merging them all in under `environment_variables` or another local if you wanted

Reply:
sounds better 👍